packages/compiler/test/shadow_css/container_queries_spec.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ShadowCss} from '@angular/compiler/src/shadow_css';
+
+describe('ShadowCss, container queries', () => {
+  function s(css: string, contentAttr: string, hostAttr: string = '') {
+    const shadowCss = new ShadowCss();
+    return shadowCss.shimCssText(css, contentAttr, hostAttr);
+  }
+
+  it('should scope normal selectors inside an unnamed container rules', () => {
+    const css = `@container max(max-width: 500px) {
+             .item {
+               color: red;
+             }
+           }`;
+    const result = s(css, 'host-a');
+    expect(noIndentation(result)).toEqual(noIndentation(`@container max(max-width: 500px) {
+         .item[host-a] {
+           color: red;
+         }
+       }`));
+  });
+
+  it('should scope normal selectors inside a named container rules', () => {
+    const css = `@container container max(max-width: 500px) {
+             .item {
+               color: red;
+             }
+           }`;
+    const result = s(css, 'host-a');
+    // Note that for the time being we are not scoping the container name itself,
+    // this is something that may or may not be done in the future depending
+    // on how the css specs evolve. Currently as of Chrome 107 it seems like shadowDom
+    // boundaries don't effect container queries (thus the scoping wouldn't be needed)
+    // And this aspect of container queries seems to be still under active discussion:
+    // https://github.com/w3c/csswg-drafts/issues/5984
```

Comment:
@JoostK this got a bit verbose :sweat: , I was wondering if I should just mention the fact but not explaining anything and just link to my comment in the issue instead?

something like:
```suggestion
    // Note that for the time being we are not scoping the container name itself,
    // for more details see: https://github.com/angular/angular/issues/48264#issuecomment-1336524296
```

what do you prefer? :slightly_smiling_face:

Reply:
I like what you have in the PR now! :+1: